internal/provider/resource_jaas_group_test.go

```diff
@@ -0,0 +1,173 @@
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGroupNameValidation(t *testing.T) {
+	testCases := []struct {
+		desc    string
+		input   string
+		matches bool
+	}{
+		{
+			desc:    "Vaild - letters",
+			input:   "foo",
+			matches: true,
+		},
+		{
+			desc:    "Vaild - letters and numbers",
+			input:   "foo123",
+			matches: true,
+		},
+		{
+			desc:    "Vaild - letters, numbers and hyphens",
+			input:   "foo-123",
+			matches: true,
+		},
+		{
+			desc:    "Vaild - letters, numbers and underscores",
+			input:   "foo_123",
+			matches: true,
+		},
+		{
+			desc:    "Vaild - letters and dots",
+			input:   "foo.bar",
+			matches: true,
+		},
+		{
+			desc:  "Invalid - starts with a number",
+			input: "1foo",
+		},
+		{
+			desc:  "Invalid - starts with a hyphen",
+			input: "-foo",
+		},
+		{
+			desc:  "Invalid - ends with a hyphen",
+			input: "foo-",
+		},
+		{
+			desc:  "Invalid - ends with special character",
+			input: "foo@",
+		},
+		{
+			desc:  "Invalid - contains special character",
+			input: "foo@bar",
+		},
+	}
+
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			assert.Equal(t, groupNameRe.MatchString(tC.input), tC.matches)
+		})
+	}
+}
+
+func TestAcc_ResourceJaasGroup(t *testing.T) {
+	OnlyTestAgainstJAAS(t)
+	groupName := acctest.RandomWithPrefix("tf-jaas-group")
+	newGroupName := acctest.RandomWithPrefix("tf-jaas-group-new")
+	resourceName := "juju_jaas_group.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: frameworkProviderFactories,
+		CheckDestroy:             testAccCheckJaasGroupDestroyed,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceJaasGroup(groupName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", groupName),
+					resource.TestCheckResourceAttrSet(resourceName, "uuid"),
+					testAccCheckJaasGroupExists(resourceName),
+				),
+			},
+			{
+				Config: testAccResourceJaasGroup(newGroupName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", newGroupName),
+					resource.TestCheckResourceAttrSet(resourceName, "uuid"),
+					testAccCheckJaasGroupExists(resourceName),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_ResourceJaasGroupInvalid(t *testing.T) {
+	OnlyTestAgainstJAAS(t)
+	invalidGroupName := "_invalid name"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: frameworkProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceJaasGroup(invalidGroupName),
+				// Might break if the formatting changes
+				ExpectError: regexp.MustCompile("must start with a letter, end with a letter or number"),
+			},
+		},
+	})
+}
+
+func testAccResourceJaasGroup(name string) string {
+	return fmt.Sprintf(`
+resource "juju_jaas_group" "test" {
+  name = %q
```

Comment:
Have a look at my PR again, I switched to using some templatting function which we can use here too.